src/player/player.rs

```diff
@@ -240,15 +242,50 @@ async fn create_session(
     credentials: Credentials,
     ap_port: Option<u16>,
 ) -> Result<Session, SpotifyError> {
-    let session_config = SessionConfig {
-        ap_port,
-        ..Default::default()
-    };
-    let result = Session::connect(session_config, credentials, None).await;
-    result.map_err(|e| {
-        dbg!(e);
-        SpotifyError::LoginFailed
-    })
+    match ap_port {
+        Some(ap_port) => {
+            let session_config = SessionConfig {
+                ap_port: Some(ap_port),
+                ..Default::default()
+            };
+            let result = Session::connect(session_config, credentials, None).await;
+            result.map_err(|e| {
+                dbg!(e);
+                SpotifyError::LoginFailed
+            })
+        },
+        None => {
+            let mut ret = None;
+
+            let mut ports: Vec<Option<u16>> = KNOWN_AP_PORTS.iter().map(|port| Some(port.to_owned())).collect();
+            ports.push(None);
+            for port in ports {
+                let session_config = SessionConfig {
+                    ap_port: port,
+                    ..Default::default()
+                };
+                let result = Session::connect(session_config, credentials.clone(), None).await;
+
+                match result {
+                    Ok(session) => {
+                        ret = Some(Ok(session));
```

Comment:
why do you have to wrap the result in an Option? When is it a None?

(also I think you can `break Ok(session)` for instance, and that would be the value of your match block)

Reply:
I meant `ret` is None if there is not any access point available, but I just replaced `ret` with `return` and this looks better (see the latest commit).

The compiler [told](https://doc.rust-lang.org/stable/reference/expressions/loop-expr.html#break-and-loop-values) me you can break with values in `loop {...}` and breakable blocks, but not inside `for`-loop.